llmfoundry/interfaces/callback_with_config.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+import abc
+from typing import Any
+
+from composer.core import Callback
+
+__all__ = ['CallbackWithConfig']
+
+
+class CallbackWithConfig(Callback, abc.ABC):
+
+    def __init__(self, config: dict[str, Any], *args: Any,
+                 **kwargs: Any) -> None:
+        del config, args, kwargs
+        pass
```

Comment:
Why is this necessary?

Reply:
We have some callbacks that just take kwargs, and some callbacks that also need access to the full train config passed in. This is just to define the interface explicitly instead of implicitly. Let me add a docstring.